patches/api/0225-Add-a-way-to-get-translation-keys-for-blocks-entitie.patch

```diff
@@ -78,36 +108,79 @@ index 786b8011e98b2fe93cc2418d624f6350ede62d90..024deba760c41787190d20e4ac5c5419
 +     * Return the translation key for the Block, so the client can translate it into the active
 +     * locale when using a TranslatableComponent.
 +     * @return the translation key
++     * @deprecated use {@link #translationKey()}
 +     */
 +    @NotNull
++    @Deprecated
 +    String getTranslationKey();
++
++    /**
++     * Return the translation key for the Block, so the client can translate it into the active
++     * locale when using a {@link net.kyori.adventure.text.TranslatableComponent}.
++     * @return the translation key
++     */
++    @Override
++    @NotNull String translationKey();
      // Paper end
  }
 diff --git a/src/main/java/org/bukkit/entity/EntityType.java b/src/main/java/org/bukkit/entity/EntityType.java
-index 9be5371c7f398d0ec8241403661415ff40661067..f415b61b0d4b57e1557aaf240a0f2ad5915035fc 100644
+index 9be5371c7f398d0ec8241403661415ff40661067..a98fb83155d49ed17f67c21aaa80140c20e7f79f 100644
 --- a/src/main/java/org/bukkit/entity/EntityType.java
 +++ b/src/main/java/org/bukkit/entity/EntityType.java
-@@ -419,4 +419,15 @@ public enum EntityType implements Keyed {
+@@ -20,7 +20,7 @@ import org.jetbrains.annotations.Contract;
+ import org.jetbrains.annotations.NotNull;
+ import org.jetbrains.annotations.Nullable;
+ 
+-public enum EntityType implements Keyed {
++public enum EntityType implements Keyed, net.kyori.adventure.translation.Translatable { // Paper - translatable
+ 
+     // These strings MUST match the strings in nms.EntityTypes and are case sensitive.
+     /**
+@@ -419,4 +419,30 @@ public enum EntityType implements Keyed {
      public boolean isAlive() {
          return living;
      }
-+
++    // Paper start
 +    /**
 +     * Return the translation key for the EntityType, so the client can translate it into the active
 +     * locale when using a TranslatableComponent.<br>
 +     * This is <code>null</code>, when the EntityType isn't known to NMS (custom entities)
 +     * @return the translation key
++     * @deprecated use {@link #translationKey()}
 +     */
++    @Deprecated
 +    @Nullable
-+    String getTranslationKey() {
++    public String getTranslationKey() {
++        return org.bukkit.Bukkit.getUnsafe().getTranslationKey(this);
++    }
++
++    /**
++     * Return the translation key for the EntityType, so the client can translate it into the active
++     * locale when using a {@link net.kyori.adventure.text.TranslatableComponent}.<br>
++     * This is <code>null</code>, when the EntityType isn't known to NMS (custom entities)
```

Comment:
```suggestion
+     * locale when using a {@link net.kyori.adventure.text.TranslatableComponent}.
+     * <p>
+     * This is {@code null} when the EntityType isn't known to NMS (custom entities)
```

Reply:
The Javadoc is wrong here, you're throwing an exception not returning null (which would be correct as the API requires the return to not be null).